homeassistant/components/thermostat/__init__.py

```diff
@@ -301,22 +306,32 @@ def turn_fan_off(self):
     @property
     def min_temp(self):
         """Return the minimum temperature."""
-        return convert(7, TEMP_CELSIUS, self.unit_of_measurement)
+        unit = self.unit_of_measurement
+        if not unit:
+            unit = self.hass.config.units.temperature_unit
+        return convert(7, TEMP_CELSIUS, unit)
 
     @property
     def max_temp(self):
         """Return the maximum temperature."""
-        return convert(35, TEMP_CELSIUS, self.unit_of_measurement)
+        unit = self.unit_of_measurement
+        if not unit:
+            unit = self.hass.config.units.temperature_unit
+        return convert(35, TEMP_CELSIUS, unit)
 
     def _convert_for_display(self, temp):
         """Convert temperature into preferred units for display purposes."""
-        if temp is None:
-            return None
+        if temp is None or not isinstance(temp, Number):
+            return temp
 
-        value = convert(temp, self.unit_of_measurement,
-                        self.hass.config.temperature_unit)
+        if self.unit_of_measurement:
+            value = \
+                self.hass.config.units.temperature(
+                    temp, self.unit_of_measurement)
+        else:
+            value = temp
 
-        if self.hass.config.temperature_unit is TEMP_CELSIUS:
+        if self.hass.config.units.temperature_unit is TEMP_CELSIUS:
```

Comment:
btw data comparison should always be done using `==`. `is` will compare object referential equality.

Also, `is_metric`  ? 💃